### DISCUSSION TITLE
Docusaurus prism-react-renderer themes

### DISCUSSION THREAD

**Discussion by userdocs** [daltonmenezes/aura-theme#177] [Ideas]
So it turns out prism-react-renderer, which is what Docusaurus uses to render code blocks, [has a tool to convert vscode themes.](https://github.com/FormidableLabs/prism-react-renderer/tree/master/tools/themeFromVsCode)

I tweaked the tool and made a Github workflow to create working Docusaurus themes.

https://github.com/userdocs/docusaurus-vscode-to-prism/tree/main/docusaurus-themes/src/prism/themes/aura

![2023-02-16_20-19-28](https://user-images.githubusercontent.com/16525024/219478848-7f195b96-ad6b-43a9-babe-f2d1424e4e49.png)

Think these are official port worthy? I used the official tool to port them.

**Comment by daltonmenezes** [comment]
Hi @userdocs , sure! It looks great for me and would be awesome to make them official! 💜

**Reply by userdocs** [reply]
So, prism updated and dropped the tool, I can still use it via my repo but I noticed the output was pretty weak with limited highlighting. hardly doing anything. This shame since the vscode json it is converted from seems very complete.

So took another look.

Starting with their included themes https://github.com/FormidableLabs/prism-react-renderer/tree/master/packages/prism-react-renderer/src/themes I thought i'd try to make a manual one just brute forcing my way through until i got the look i wanted across a range of languages.

This appears the most complete theme working across most langs I tested (to basically copy and change the colours)

https://github.com/FormidableLabs/prism-react-renderer/blob/master/packages/prism-react-renderer/src/themes/synthwave84.ts

So far I have this 

https://github.com/userdocs/docusaurus-aura-theme/blob/main/src/prism/themes/aura/aura-custom.ts

You can see it working here

https://userdocs.github.io/docusaurus-aura-theme/docs/bash

I'm not really JS proficient here to automate this or do it in js smart way. If we could get maybe 1 or two complete prism react themes we  could maybe pr them to be included in the official repo?

Aura is the best theme.
